pkg/dio/multi.go

```diff
@@ -192,6 +192,11 @@ func (t *multiReader) Read(p []byte) (int, error) {
 		}
 	}
 
+	// return 0, nil if there are no readers
+	if len(t.readers) == 0 {
```

Comment:
@hmahmood t.readers can be modified by Remove - this will likely occur while Read is blocked on line 182.